tests/diagnostics/uninitialized-out.slang

```diff
@@ -2,6 +2,56 @@
 
 float foo(out float3 v)
 {
+    // This should error as we haven't set v before we read from it
     float r = v.x + 1.0;
+    // This should warn as we haven't set v before we return
     return r;
 }
+
+// This should warn as we return without x being initialized
+float bar(out float x)
+{
+    return 0;
+}
+
+// This should also warn pointing at the implicit return
+void baz(out float x) {}
+
+void twoReturns(bool b, out float y)
+{
+    if(b)
+    {
+        // Should warn
+        return;
+    }
+    y = 0;
+    // Shouldn't warn
+    return;
+}
+
+void twoOkReturns(bool b, out float y)
+{
+    if(b)
+    {
+        // Shouldn't warn
+        unused(y);
+        return;
+    }
+    y = 0;
+    // Shouldn't warn
+    return;
+}
+
+// TODO: This should warn that n is potentially uninitialized
```

Comment:
The tricky case is something like
```
void f(out float v)
{
     for (int i = 0; i < 5; i++)
     {
          if (i == 2) v = 5;
     }
    return; // should not warn.
}
```

Reply:
indeed, there are only so many cases which we can reasonably catch. I think this is definitely one of them though